internal/sbi/consumer/nrf_service.go

```diff
@@ -84,60 +84,47 @@ func (s *nnrfService) SendSearchNFInstances(
 		return nil, openapi.ReportError("nrf not found")
 	}
 
-	ctx, _, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NNRF_DISC, models.NfType_NRF)
+	ctx, _, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NNRF_DISC, models.NrfNfManagementNfType_NRF)
 	if err != nil {
 		return nil, err
 	}
 
-	result, res, err := client.NFInstancesStoreApi.SearchNFInstances(ctx, targetNfType, requestNfType, param)
+	res, err := client.NFInstancesStoreApi.SearchNFInstances(ctx, &param)
 
-	if res != nil && res.StatusCode == http.StatusTemporaryRedirect {
-		return nil, fmt.Errorf("temporary Redirect For Non NRF Consumer")
-	}
-	if res == nil || res.Body == nil {
-		return &result, err
+	if err != nil || res == nil {
+		logger.ConsumerLog.Errorf("SearchNFInstances failed: %+v", err)
+		return nil, err
 	}
-	defer func() {
-		if res != nil {
-			if bodyCloseErr := res.Body.Close(); bodyCloseErr != nil {
-				err = fmt.Errorf("SearchNFInstances' response body cannot close: %+w", bodyCloseErr)
-			}
-		}
-	}()
+
+	result := res.SearchResult
 	return &result, err
 }
 
 func (s *nnrfService) SendDeregisterNFInstance() (problemDetails *models.ProblemDetails, err error) {
 	logger.ConsumerLog.Infof("Send Deregister NFInstance")
 
-	ctx, pd, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NNRF_NFM, models.NfType_NRF)
+	ctx, pd, err := ausf_context.GetSelf().GetTokenCtx(models.ServiceName_NNRF_NFM, models.NrfNfManagementNfType_NRF)
 	if err != nil {
 		return pd, err
 	}
 
 	ausfContext := s.consumer.Context()
 	client := s.getNFManagementClient(ausfContext.NrfUri)
 
-	var res *http.Response
+	request := &Nnrf_NFManagement.DeregisterNFInstanceRequest{
+		NfInstanceID: &ausfContext.NfId,
+	}
 
-	res, err = client.NFInstanceIDDocumentApi.DeregisterNFInstance(ctx, ausfContext.NfId)
-	if err == nil {
-		return problemDetails, err
-	} else if res != nil {
-		defer func() {
-			if resCloseErr := res.Body.Close(); resCloseErr != nil {
-				logger.ConsumerLog.Errorf("DeregisterNFInstance response cannot close: %+v", resCloseErr)
-			}
-		}()
-		if res.Status != err.Error() {
-			return problemDetails, err
-		}
-		problem := err.(openapi.GenericOpenAPIError).Model().(models.ProblemDetails)
-		problemDetails = &problem
-	} else {
-		err = openapi.ReportError("server no response")
+	_, err = client.NFInstanceIDDocumentApi.DeregisterNFInstance(ctx, request)
+
+	switch e := err.(type) {
+	case openapi.GenericOpenAPIError:
+		return e.Model().(*models.ProblemDetails), nil
```

Comment:
`ProblemDetails` is an element of `DeregisterNFInstanceError`.

Reply:
FYR 
https://github.com/free5gc/udm/pull/38/files#diff-b3e57065c4b3465aacc7853836b2b68e4b64fa16167f38722a90772d157d6e0b